src/components/registration/LogIn.js

```diff
@@ -26,53 +36,91 @@ class LogIn extends Component {
       .post(`${REACT_APP_SVR_API}/user/login`, data)
       .then(response => {
         localStorage.setItem('app-token', response.data.token);
-        // this.props.history.push("/dashboard");
-        this.props.handleLogin();
+        handleLogin();
       })
       .catch(error => {
-        try {
-          // Handles errors that are not HTTP specific
-          console.error(error);
-          this.setState({ showRegistrationFailure: true });
-          if (!error.status) {
-            console.error('A network error has occured.');
-          } else if (error.response.status === 400) {
-            console.error('Bad Request');
-          } else if (error.response.status === 500) {
-            console.error('Something bad happended on the server.');
-          } else {
-            console.error('An unknown error has occurred');
-          }
-        } catch (ex) {
-          Promise.reject(ex);
+        if (error.message === 'Network Error') {
+          this.setState({
+            HTTPErrorMessage:
+              'A network error has occurred while contacting our servers...',
+          });
+        } else if (error.response.status === 400) {
+          this.setState({ HTTPErrorMessage: `${error.response.data}` });
+        } else if (error.response.status === 500) {
+          this.setState({
+            HTTPErrorMessage: `Something went wrong on our side. Please try again at a later time.\n Error: ${error.response.data}`,
+          });
         }
       });
   };
 
-  render() {
+  handleEmailChange = evt => {
+    this.setState({ email: evt.target.value });
+  };
+
+  handlePasswordChange = evt => {
+    this.setState({ password: evt.target.value });
+  };
+
+  validateData = () => {
     const { email, password } = this.state;
+    let setErrors = { email: false, password: false };
+    let atLeastOneFailed = false;
+    // Validating Email
+    if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) {
+      setErrors.email = true;
+      atLeastOneFailed = true;
+    }
+    // Validating password
+    if (password.length === 0 || password === undefined) {
+      setErrors.password = true;
+      atLeastOneFailed = true;
+    }
+    // Setting error flags in state
+    if (!atLeastOneFailed) {
```

Comment:
Use array methods to just confirm that all values in your `setErrors` are false:

```suggestion
    if (Object.keys(setErrors).every(err_type => !setErrors[err_type])) {
```